scripts/prepare.js

```diff
@@ -68,21 +64,34 @@ function logError(type, packageJson, errorLogs) {
   );
 }
 
-const modulePath = path.resolve('./');
-const packageJson = getPackageJson(modulePath);
 const modules = true;
 
-async function prepare() {
-  removeDist();
+async function prepare({ cwd, flags }) {
+  const { packageJson } = await readPkgUp(cwd);
+  const message = chalk.gray(`Built: ${chalk.bold(`${packageJson.name}@${packageJson.version}`)}`);
+  console.time(message);
 
-  await babelify({
-    modules,
-    errorCallback: (errorLogs) => logError('js', packageJson, errorLogs),
-  });
-  tscfy({ errorCallback: (errorLogs) => logError('ts', packageJson, errorLogs) });
+  if (flags.includes('--reset')) {
+    await removeDist();
+  }
 
-  cleanup();
-  console.log(chalk.gray(`Built: ${chalk.bold(`${packageJson.name}@${packageJson.version}`)}`));
+  await Promise.all([
+    babelify({
+      modules,
+      watch: flags.includes('--watch'),
+      errorCallback: (errorLogs) => logError('js', packageJson, errorLogs),
+    }),
+    tscfy({
+      watch: flags.includes('--watch'),
+      errorCallback: (errorLogs) => logError('ts', packageJson, errorLogs),
+    }),
```

Comment:
Let's do them both in parallel, why not? their outputs might mingle, but the only output we're expecting are errors!

Reply:
They currently still run in sub-processes anyway